Assets/Scripts/Enemy/EnemyData.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Bug.Enemy
+{
+    public class EnemyData : MonoBehaviour
+    {
+        [SerializeField]
+        public DamageModifier[] _damageModifier;
+
+        private Dictionary<Collider, float> _internalModifier;
+
+        private void Start()
+        {
+            _internalModifier = _damageModifier.ToDictionary(x => x.Collider, x => x.Modifier);
+        }
+
+        private int _health = 100;
+
+        public int TakeDamage(Collider collider, int damage)
```

Comment:
Should the damage be a float? Would allow us more flexibility later, without constraining us now!

Reply:
Will do

Reply:
Fixed with d9c7776ffb7344f259345b745b22e6977836e19e